### PR TITLE
Update first React class

### PR DESCRIPTION
**Pull request by llh1**
Few fixes based on our experience in Scotland CYF:

- We taught students to write JS with ';' at the end of each line, and create-react-app also uses this convention, so exercises and examples have been updated to use this as well.
- Use `querySelector` over `getElementById`.
- Typos and various small fixes

### CODE REVIEW

File: react/week-19/lesson.md [hunk 1]
```diff
@@ -135,8 +134,7 @@ Using JSX ([interactive version](http://jsbin.com/gekahexige/edit?html,output)):
 As you can see, this is much easier to read than both the straight `React.createElement` API and the vanilla JS API. Most people using React use JSX to write their components.
 
 > **Exercise B**
-> 1. Change the [JSX example from above](http://jsbin.com/gekahexige/edit?html,output) to instead render a `h1` with an id of `"main-header"` and the text "Hello Ali"
-> 2. Visit the [Babel REPL](https://babeljs.io/repl/) and copy/paste the `h1` that you just created
```
Comment (alismith-dev) [inline react/week-19/lesson.md:139]: Perhaps it's just me, but I do quite like the ability for students to see that JSX isn't "magic", it's just regular functions. I feel like it reduces the "this is weird new syntax" aspect of it.

Having said that I felt like we spent too much time on it in London for the last class. So I think I'm ok with removing this exercise.

File: react/week-19/lesson.md [hunk 2]
```diff
@@ -159,27 +157,28 @@ The last command should open a web browser for you. This shows the application t
...
 ```
 
-There are 3 important parts in this code:
+There are 4 important parts in this code:
 
-1. First we import `React`. This is important because JSX is converted to `React.createElement` calls. If the `React` variable is undefined then this will fail
-2. We create a React component called `HelloWorld`
-3. We *render* the `HelloWorld` component into a `div` with the id of `root`
+1. First we import `React`. This is important because JSX is converted to `React.createElement` calls. If the `React` variable is undefined then this will fail.
+2. We import `ReactDOM` which will be used to *render* the component on the screen.
```
Comment (alismith-dev) [inline react/week-19/lesson.md]: Interestingly `ReactDOM` kept coming up as a problem for class 5 in London.

In all of our interactive examples we have the `ReactDOM.render` right in the same file. Which is different to the student's CRA code - there's already a `ReactDOM.render` implemented in `index.js`.

They thought that it was important to have `ReactDOM.render` in every file. Additionally we use the word "render" a lot to mean returning a component, so it kind of makes sense that you would need a method named `render`.

I found it difficult to explain what `ReactDOM.render` was doing. And how it was different from just regular `React` without going off on a major tangent.

I'm leaning towards mostly ignoring it and changing all of the examples to have a split between the code that mounts onto the root node and the "application" code. Basically to mimic the default CRA file structure.

File: react/week-19/lesson.md [hunk 3]
```diff
@@ -159,27 +157,28 @@ The last command should open a web browser for you. This shows the application t
 We looked at the beginning of the lesson at the concept of components. Now let's look at how components are made in React.
 
 ```js
-import React from 'react'
-import ReactDOM from 'react-dom'
+import React from 'react';
+import ReactDOM from 'react-dom';
 
 function HelloWorld() {
   return (
```
Comment (alismith-dev) [inline react/week-19/lesson.md:164]: Just a quick note on the parens here - I was deliberately trying to leave them in, because multi-line JSX can cause problems sometimes. 

I just tested this:

```js
return
  <div>
    <span>Hello world</span>
  </div>
```

Prettier will insert a semi after the return, which I think would trip up the students.

However this for this code:

```js
return <div>
  <span>Hello world</span>
</div>
```

Prettier does the right thing.  I feel like this would be the more common scenario for students to type, so in the end I don't think it really matters that much.

Overall I think it's something that's good to mention when showing a component for the first time.

File: react/week-19/lesson.md [hunk 4]
```diff
@@ -189,19 +188,15 @@ There are 3 important parts in this code:
 
 #### Component Composition
 
-A component can be combined with another component so that both are rendered. This is sometimes called *composition* ([interactive example](https://codesandbox.io/s/0x4wonqn00)):
+A component can be combined with another component so that both are rendered. This is called *composition* ([interactive example](https://codesandbox.io/s/0x4wonqn00)):
```
Comment (alismith-dev) [inline react/week-19/lesson.md:190]: I can get these codesandboxes updated with those tweaks.

I suppose we could probably set up a group account or something so that they're not all under my account.

Also, while I'm on the subject - the codesandboxes are all configured to be "frozen", which means that if you edit them, and accidentally hit save then it will prompt you to fork. I find it **really** useful to be able to open up live examples and play around with them, but I hit save out of habit so now I won't mess up the original example :)

File: react/week-19/lesson.md [hunk 5]
```diff
@@ -332,57 +338,55 @@ Here we are using `Array.map` to turn an array of strings into an array of compo
...
-      )
+      return <li key={index}>{name}</li>;
     })}
   </ul>
-)
+);
 ```
 
-Here we have added a `key` prop the to element. A documentation page explaining in more depth is in the further reading section but basically the `key` prop has a special meaning in React because it is used internally to keep track of which element in the list is which.
+Here we have added a `key` property to the `li` element. A documentation page explaining in more depth is in the further reading section but basically the `key` prop has a special meaning in React because it is used internally to keep track of which element in the list is which.
```
Comment (alismith-dev) [inline react/week-19/lesson.md]: My feeling generally we refer to `key` as a "prop", so I think for consistency I'd prefer to keep that.

File: react/week-19/lesson.md [hunk 6]
```diff
@@ -399,55 +403,53 @@ The convention is to name component files the exactly same as the component (inc
 
 ## Making an Argument for Props
 
-What's the problem with our `HelloMentor` component above? Hint: imagine what our boss might ask for with this small application. What could our boss ask for which would mean we would have to make changes to the code?
```
Comment (alismith-dev) [inline react/week-19/lesson.md:402]: I think this a good call to remove this from the syllabus text (I can get a bit wordy, sorry!).

However I have found that it is useful to contextualise examples in a "real world" example like a job. I've observed that our students don't really have a good idea of what a development job looks like on a daily basis. Maybe it's worth giving a story like this in the class?

**Review by alismith-dev — COMMENTED** [review]
This is really great! Thanks @llh1! I've left a few comments/questions :)

**Comment by alismith-dev** [comment]
> We taught students to write JS with ';' at the end of each line, and create-react-app also uses this convention, so exercises and examples have been updated to use this as well.

Sorry! I don't use semis in my work (and personal projects), so it's just habit on my part. I try to fix that when I see it as we do teach semis in London too. Thanks for fixing those.

In terms of the CodeSandbox links, we should probably work out something so that we can collectively edit them. It's a bit of a pain to have to update the links in the syllabus if we change 1 minor thing. In the meantime I can go through and update all of the existing examples to use semis, if you think that is preferable.

File: react/week-19/lesson.md (hunk 1 above)
Comment (llh1) [reply react/week-19/lesson.md:139]: Right i understand, I think mentor can maybe just show to the class that actually JSX is just Javascript finally instead of having it as an exercise? What do you think? Could add it as a mentor note.

File: react/week-19/lesson.md (hunk 6 above)
Comment (llh1) [reply react/week-19/lesson.md:402]: Yes it's what we tend to do in Glasgow CYF, we don't necessarily stick to the syllabus when explaining or demonstrating things, we add some examples or stories that we say to the class which are not part of the syllabus. I think the syllabus can be seen as a reference for the theory but after it's up to each mentor to use the right examples to let students understand a concept. Maybe this could be worth adding to the mentor notes instead?

**Review by alismith-dev — APPROVED** [review]
Thanks for making those changes!

As discussed on slack, I'll transfer my existing CodeSandboxes to a CYF team and make sure that they get correctly prettified.

File: react/week-19/lesson.md (hunk 1 above)
Comment (alismith-dev) [reply react/week-19/lesson.md:139]: I think adding it as a mentor note is a good idea 👍

**Comment by alismith-dev** [comment]
@llh1 I've now transferred all of the CodeSandbox examples to a CYF team and prettified them. You have have an invite to the CodeSandbox team

File: react/week-19/lesson.md (hunk 6 above)
Comment (alismith-dev) [reply react/week-19/lesson.md:402]: I think adding to the mentor notes is good - I've found that having concrete examples is good reference for new mentors

**Review by alismith-dev — APPROVED** [review]
Great! Thanks for that @llh1!
